macro-analytics/Code.js

```diff
@@ -0,0 +1,560 @@
+    spreadsheet.getCurrentCell().setFormula('=TODAY()-WEEKDAY(TODAY())-1');
+    spreadsheet.getRange('C3').activate();
+    spreadsheet.getRange('B3:B7').copyTo(spreadsheet.getActiveRange(), SpreadsheetApp.CopyPasteType.PASTE_NORMAL, false);
+    spreadsheet.getRange('C2').activate();
+    spreadsheet.getCurrentCell().setValue('Last week');
+    spreadsheet.getRange('C5').activate();
+    spreadsheet.getCurrentCell().setFormula('=B4-1');
+    spreadsheet.getRange('C4').activate();
+    spreadsheet.getCurrentCell().setFormula('=B4-7');
+    spreadsheet.getRange('D3').activate();
+    spreadsheet.getRange('B3:C7').copyTo(spreadsheet.getActiveRange(), SpreadsheetApp.CopyPasteType.PASTE_NORMAL, false);
+    spreadsheet.getRange('D2').activate();
+    spreadsheet.getCurrentCell().setValue('This year');
+    spreadsheet.getRange('E2').activate();
+    spreadsheet.getCurrentCell().setValue('Last year');
+    spreadsheet.getRange('D7').activate();
+    spreadsheet.getCurrentCell().setValue('ga:month');
+    spreadsheet.getRange('E7').activate();
+    spreadsheet.getCurrentCell().setValue('ga:month');
+    spreadsheet.getRange('D4').activate();
+    spreadsheet.getCurrentCell().setFormula('=DATE(YEAR(TODAY()),1,1)');
+    spreadsheet.getRange('E4').activate();
+    spreadsheet.getCurrentCell().setFormula('=D4-365');
+    spreadsheet.getRange('F3').activate();
+    spreadsheet.getRange('D3:D7').copyTo(spreadsheet.getActiveRange(), SpreadsheetApp.CopyPasteType.PASTE_NORMAL, false);
+    spreadsheet.getRange('F2').activate();
+    spreadsheet.getCurrentCell().setValue('Top browsers');
+    spreadsheet.getRange('F7').activate();
+    spreadsheet.getCurrentCell().setValue('ga:browser');
+    spreadsheet.getRange('F8').activate();
+  };
+  
+  
+  function createSummary() {
+   var spreadsheet = SpreadsheetApp.getActive();
+    spreadsheet.getRange('B1').activate();
+    spreadsheet.getCurrentCell().setValue('This week');
+    spreadsheet.getRange('C1').activate();
+    spreadsheet.getCurrentCell().setValue('Last week');
+    spreadsheet.getRange('A2').activate();
+    spreadsheet.getCurrentCell().setValue('Sun');
+    spreadsheet.getActiveRange().autoFill(spreadsheet.getRange('A2:A8'), SpreadsheetApp.AutoFillSeries.DEFAULT_SERIES);
+    spreadsheet.getRange('B2').activate();
+    spreadsheet.getCurrentCell().setFormula('=\'This week\'!B16');
+    spreadsheet.getRange('C2').activate();
+    spreadsheet.getCurrentCell().setFormula('=\'Last week\'!B16');
+    spreadsheet.getRange('B2').activate();
+    spreadsheet.getActiveRange().autoFill(spreadsheet.getRange('B2:B8'), SpreadsheetApp.AutoFillSeries.DEFAULT_SERIES);
+    spreadsheet.getRange('C2').activate();
+    spreadsheet.getActiveRange().autoFill(spreadsheet.getRange('C2:C8'), SpreadsheetApp.AutoFillSeries.DEFAULT_SERIES);
+    spreadsheet.getRange('A1:C8').activate();
+    var sheet = spreadsheet.getActiveSheet();
+    var chart = sheet.newChart()
+    .asColumnChart()
```

Comment:
This was probably generated from the macro as is, but all the lines after `var chart = ...` starting with a dot (`.`) are chained methods from the chart, so they're all part of that same instruction. It's easier to read if those lines are indented to show that.

```js
var chart = sheet.newChart()
    .asColumnChart()
    .addRange(...
    ...
    .build();
sheet.insertChart(chart);
...
```

If you want to show the code exactly as it's created from the macro recorder, maybe leave it like this. It's up to you if you want to change it.

Reply:
Good call, added!